internal/internal_task_handlers.go

```diff
@@ -1005,19 +1022,33 @@ ProcessEvents:
 		}
 		reorderedEvents := nextTask.events
 		markers := nextTask.markers
-		historyMessages := nextTask.msgs
+		historyMessages := nextTask.acceptedMsgs
 		flags := nextTask.flags
 		binaryChecksum := nextTask.binaryChecksum
 		nextTaskBuildId := nextTask.buildID
+		admittedUpdates := nextTask.admittedMsgs
 		// Check if we are replaying so we know if we should use the messages in the WFT or the history
 		isReplay := len(reorderedEvents) > 0 && reorderedHistory.IsReplayEvent(reorderedEvents[len(reorderedEvents)-1])
 		var msgs *eventMsgIndex
 		if isReplay {
+			requestUpdateBodies := make(map[string]*anypb.Any)
```

Comment:
```suggestion
			requestUpdateBodies := make(map[string]*anypb.Any, len(admittedUpdates))
```

(pedantic, but a good habit)